src/wallet/wallet_rpc_server.cpp

```diff
@@ -292,15 +295,33 @@ namespace tools
     entry.fee = 0; // TODO
     entry.note = m_wallet->get_tx_note(pd.m_tx_hash);
     entry.type = "pool";
+    entry.subaddr_index = pd.m_subaddr_index;
   }
   //------------------------------------------------------------------------------------------------------------------------------
   bool wallet_rpc_server::on_getbalance(const wallet_rpc::COMMAND_RPC_GET_BALANCE::request& req, wallet_rpc::COMMAND_RPC_GET_BALANCE::response& res, epee::json_rpc::error& er)
   {
     if (!m_wallet) return not_open(er);
     try
     {
+      res.current_subaddress_account = m_wallet->get_current_subaddress_account();
       res.balance = m_wallet->balance();
       res.unlocked_balance = m_wallet->unlocked_balance();
+      std::map<uint32_t, uint64_t> balance_per_subaddress = m_wallet->balance_per_subaddress();
+      std::map<uint32_t, uint64_t> unlocked_balance_per_subaddress = m_wallet->unlocked_balance_per_subaddress();
+      std::vector<tools::wallet2::transfer_details> transfers;
+      m_wallet->get_transfers(transfers);
+      for (auto& i : balance_per_subaddress)
+      {
+        wallet_rpc::COMMAND_RPC_GET_BALANCE::per_subaddress_info info;
+        info.index = i.first;
+        info.address = m_wallet->get_subaddress_as_str(info.index);
+        info.balance = i.second;
+        info.unlocked_balance = unlocked_balance_per_subaddress[i.first];
+        info.label = m_wallet->get_subaddress_label(i.first);
```

Comment:
It's just for providing some additional information. The same goes for the address RPC command. I intended to make the PRC commands similar to the CLI commands (i.e. you get to see address labels via `address` and `balance detail` commands). On the other hand, there's no "get account/address label" RPC command; there are only "set account/address label" commands. Do you think it's confusing?